plugins/myfitnesspal.py

```diff
@@ -2,21 +2,33 @@
 import math
 import requests
 from bs4 import BeautifulSoup
+from dateutil import parser
 
 from cloudbot import hook
 
-scrape_url = "http://www.myfitnesspal.com/food/diary/{}"
+scrape_url = "http://www.myfitnesspal.com/food/diary/{}?date={}"
 
 
 @hook.command('mfp', 'myfitnesspal')
 def mfp(text, bot):
-    """<user> - returns macros from the MyFitnessPal food diary of <user>"""
-    request = requests.get(scrape_url.format(text))
+    """
+    <user> [date]- returns macros from the MyFitnessPal food diary of <user>
+                   optionally, specify [date] to retrieve that day's diary
+    """
+    date = 'today'
+
+    args = text.split()
+    user = args[0]
+    if(len(args) > 1):
+        dt = parser.parse(' '.join(args[1:]))
```

Comment:
To go along with my comment about the `.split()`, you wouldn't need to use a join if the text is only split on the first space.